### ISSUE TITLE
Use auto generation for generating the api

### ISSUE THREAD

**Issue by farodin91** [fristonio/docker.rs#3]
Cool Project, I'm currently trying the same stuff. I try to use swagger codegen for generating the docker api.

I would like to help on this topic.

I found another project which uses go to generated the a docker api: https://github.com/Microsoft/Docker.DotNet/tree/master/tools/specgen

**Comment by fristonio** [comment]
This would be great, can you send a pull request regarding the same. :)

**Comment by farodin91** [comment]
I'm working on it. The generated models have arround 400 errors. Mostly three or four types of errors.

**Comment by fristonio** [comment]
Great, Looking forward to the Pull Request. In the meantime, I am working on `images` endpoints for docker. I need them in one of my projects :P

**Comment by frol** [comment]
FYI, [Microsoft announced their use of Rust in their Iotedge project](https://www.reddit.com/r/rust/comments/8ub964/microsoft_announces_using_rust_to_build_some_of/?st=JIY37FXH&sh=fbf511fc), and it turns out that they have a generated docker-rs crate there: https://github.com/Azure/iotedge/tree/master/edgelet/docker-rs and there are some details about its state: https://github.com/Azure/iotedge/commit/a432894e2d1eb3627e416730de88f5aa643433e6

**Comment by fristonio** [comment]
@frol though it does look good, I don't think it's published as a crate.
Also, there is literally no documentation whatsoever. I think they were facing the same problem, no stable and mature library for docker interaction so they decided to develop their own. There are stable docker wrappers in other language like python but all the crates rust have are not complete and lacks some important features.

**Comment by frol** [comment]
> I don't think they published it as a crate.

You are right, they didn't, but: https://github.com/Azure/iotedge/issues/1

> so they decided to develop their own.

The only minor detail here I want to add is that they didn't develop it from scratch, they used Swagger-Codegen (with some patches mentioned in the referenced commit message) to generate the crate.

**Comment by farodin91** [comment]
It would be cool if microsoft would move this crate into a own repo.
